### PR TITLE
PR template additions

### PR DESCRIPTION
**Pull request by JanneKiiskila**
Board PRs require a few more checks.

[x] I confirm this contribution is my own and I agree to license it with Apache 2.0.
[x] I confirm the moderators may change the PR before merging it in.

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -7,6 +7,11 @@ For more information on the requirements for pull requests, please see [the CONT
 [] I confirm this contribution is my own and I agree to license it with Apache 2.0.
 [] I confirm the moderators may change the PR before merging it in.
```
Comment (MarceloSalazar) [inline .github/pull_request_template.md]: Can we add an extra line to help to visualize this?

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -7,6 +7,11 @@ For more information on the requirements for pull requests, please see [the CONT
...
 
+For new board enablements only.
+[] I confirm the board is Mbed enabled and passes the Mbed enabled test set.
+[] I confirm the contribution has been tested properly and tests results for [TESTS](https://github.com/ARMmbed/mbed-os-example-pelion/tree/master/TESTS) are attached.
+[] I confirm `mbed-os.lib` and `mbed-cloud-client.lib` hashes were and/or the content in those folders was not modified in order to pass the tests.
```
Comment (MarceloSalazar) [inline .github/pull_request_template.md]: Can we add an extra lines to help to visualize this?

File: .github/pull_request_template.md [hunk 3]
```diff
@@ -7,6 +7,11 @@ For more information on the requirements for pull requests, please see [the CONT
 [] I confirm this contribution is my own and I agree to license it with Apache 2.0.
 [] I confirm the moderators may change the PR before merging it in.
 
+For new board enablements only.
+[] I confirm the board is Mbed enabled and passes the Mbed enabled test set.
```
Comment (MarceloSalazar) [inline .github/pull_request_template.md]: Please use capital letters for 'Mbed Enabled' and link to the requirements ;-)
https://www.mbed.com/en/about-mbed/mbed-enabled/introduction/

File: .github/pull_request_template.md (hunk 2 above)
Comment (JanneKiiskila) [reply .github/pull_request_template.md]: Added. GitHub shows these quite strangely.

File: .github/pull_request_template.md (hunk 3 above)
Comment (JanneKiiskila) [reply .github/pull_request_template.md]: Fixed.

File: .github/pull_request_template.md [hunk 4]
```diff
@@ -21,3 +12,21 @@ For more information on the requirements for pull requests, please see [the CONT
...
+
+For more information on the requirements for pull requests, please see [the CONTRIBUTING.md](CONTRIBUTING.md).
+
+-->
+
+[] I confirm this contribution is my own and I agree to license it with Apache 2.0.
+
+[] I confirm the moderators may change the PR before merging it in.
+
+For new board enablements only.
```
Comment (peknis01) [inline .github/pull_request_template.md]: Maybe a colon instead of full stop.

File: .github/pull_request_template.md [hunk 5]
```diff
@@ -21,3 +12,21 @@ For more information on the requirements for pull requests, please see [the CONT
...
+
+[] I confirm the board is [Mbed Enabled](https://www.mbed.com/en/about-mbed/mbed-enabled/introduction/) and passes the Mbed Enabled test set.
+
+[] I confirm the contribution has been tested properly and tests results for [TESTS](https://github.com/ARMmbed/mbed-os-example-pelion/tree/master/TESTS) are attached.
```
Comment (peknis01) [inline .github/pull_request_template.md]: ... and the test results

File: .github/pull_request_template.md [hunk 6]
```diff
@@ -21,3 +12,21 @@ For more information on the requirements for pull requests, please see [the CONT
...
+
+[] I confirm `mbed-os.lib` and `mbed-cloud-client.lib` hashes were and/or the content in those folders was not modified in order to pass the tests.
```
Comment (peknis01) [inline .github/pull_request_template.md]: I confirm I haven't modified the ... hashes or the content in the folders (which folders?) to pass the tests.

File: .github/pull_request_template.md (hunk 5 above)
Comment (JanneKiiskila) [reply .github/pull_request_template.md]: Fixed.

File: .github/pull_request_template.md (hunk 4 above)
Comment (JanneKiiskila) [reply .github/pull_request_template.md]: Fixed.

**Comment by JanneKiiskila** [comment]
All issues fixed (reported so far), rebased, squashed and force-pushed.

File: .github/pull_request_template.md [hunk 7]
```diff
@@ -21,3 +12,23 @@ For more information on the requirements for pull requests, please see [the CONT
...
+
+For new board enablements only:
+
+[] I confirm the board is [Mbed Enabled](https://www.mbed.com/en/about-mbed/mbed-enabled/introduction/) and passes the Mbed Enabled test set.
+
+[] I confirm the contribution has been tested properly and the tests results for [TESTS](https://github.com/ARMmbed/mbed-os-example-pelion/tree/master/TESTS) are attached.
+
+[] I confirm `mbed-os.lib` and `mbed-cloud-client.lib` hashes were or the content in folders `mbed-os` and `mbed-cloud-client` was not modified in order to pass the tests.
+- Maintainers of this repository update the Mbed OS and Client hashes.
+- Any issues those repositories should be reported via issues to those repositories.
```
Comment (peknis01) [inline .github/pull_request_template.md]: Please report any issues through issues in relevant repositories.

**Comment by JanneKiiskila** [comment]
Fixed one time. :-)

**Pull request merged**